File: src/mk1/JavaPackage/src/main/java/tamaf/CyclicBehavior.java
package tamaf;

public abstract class CyclicBehavior extends Behavior {
    
    public CyclicBehavior() {
        super();
    }

    public CyclicBehavior(Agent agent) {
        super(agent);
    }
    
    @Override
    public boolean done() { return false; }
}

File: src/mk3/esp32package/TAMAF-ESP32/src/tamaf/behaviors/CyclicBehavior.cpp
#include "CyclicBehavior.h"

namespace tamaf {
namespace behaviors {

CyclicBehavior::CyclicBehavior() : Behavior() {}

CyclicBehavior::CyclicBehavior(tamaf::core::Agent* agent) : Behavior() {
    setAgent(agent);
}

bool CyclicBehavior::Done() {
    return false; // Never finishes automatically
}

} // namespace behaviors
} // namespace tamaf
